### PR TITLE
rockchip: rk3328: support rk3328

### PR DESCRIPTION
**Pull request by TonyXie06**
rk3328 is a Quad-core soc and Cortex-a53 inside!
This patch supports the following functions:
1、power up/off cpus
2、suspend/resume cpus
3、suspend/resume system
4、reset system
5、power off system

Change-Id: I60687058d13912c6929293b06fed9c6bc72bdc84
Signed-off-by: tony.xie <tony.xie@rock-chips.com>

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
@TonyXie06 
Quick update - have taken an initial pass / review of the PR, but as it's a new platform I'd like to get a second review ahead of approving.

**Comment by davidcunado-arm** [comment]
jenkins: test this please

File: plat/rockchip/rk3328/drivers/pmu/plat_pmu_macros.S [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
```
Comment (antonio-nino-diaz-arm) [inline plat/rockchip/rk3328/drivers/pmu/plat_pmu_macros.S]: This copyright notice is missing a paragraph:

> Neither the name of ARM nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

File: plat/rockchip/common/plat_pm.c [hunk 2]
```diff
@@ -412,6 +412,7 @@ const plat_psci_ops_t plat_rockchip_psci_pm_ops = {
 	.pwr_domain_suspend = rockchip_pwr_domain_suspend,
 	.pwr_domain_on_finish = rockchip_pwr_domain_on_finish,
 	.pwr_domain_suspend_finish = rockchip_pwr_domain_suspend_finish,
+	.pwr_domain_pwr_down_wfi = rockchip_pd_pwr_down_wfi,
```
Comment (davidcunado-arm) [inline plat/rockchip/common/plat_pm.c]: @TonyXie06 
This function is missing for rk3368 and rk3399 - checks failed, e.g. for rk3368:

/debug/bl31/plat_pm.d -MT build/rk3368/debug/bl31/plat_pm.o -MP -c plat/rockchip/common/plat_pm.c -o build/rk3368/debug/bl31/plat_pm.o
plat/rockchip/common/plat_pm.c:406:29: error: 'rockchip_pd_pwr_down_wfi' undeclared here (not in a function)
  .pwr_domain_pwr_down_wfi = rockchip_pd_pwr_down_wfi,
                             ^
make: *** [build/rk3368/debug/bl31/plat_pm.o] Error 1

Comment (TonyXie06) [inline plat/rockchip/common/plat_pm.c]: I am sorry that I reply it late. I do not find this error and rockchip_pd_pwr_down_wfi() is implement in the same file with (.pwr_domain_pwr_down_wfi = rockchip_pd_pwr_down_wfi); 
This patch needs to base on https://github.com/ARM-software/arm-trusted-firmware/pull/854.

File: plat/rockchip/rk3328/drivers/pmu/plat_pmu_macros.S (hunk 1 above)
Comment (TonyXie06) [inline plat/rockchip/rk3328/drivers/pmu/plat_pmu_macros.S]: I am sorry that I reply it late. I will add it.

File: plat/rockchip/common/plat_pm.c (hunk 2 above)
Comment (antonio-nino-diaz-arm) [inline plat/rockchip/common/plat_pm.c]: Oh, I see the problem. We need to revert this change: https://github.com/ARM-software/arm-trusted-firmware/pull/889

**Comment by davidcunado-arm** [comment]
jenkins: test this please

**Comment by davidcunado-arm** [comment]
@TonyXie06 
Now that this platform port is merged, we've been able to run the static analysis tool over it and unfortunately an issue has been raised in rk3328 version of rockchip_soc_cores_pwr_dm_on.

There is a potential out-of-bounds access of cpuson_flags and cpuson_entry_point because the index cpu_id could in some circumstances be set to -1, which as a unit32_t will result in a index beyond the size of these 2 arrays.

The rk3399 version of this function solves this by having an assert for cpu_id - assert(cpu_id < PLATFORM_CORE_COUNT); 

Please can you provide a fix for rk3328?

**Comment by TonyXie06** [comment]
Thank you for your check， I will modify this。
